comprehensive_microservice/coder.py

```diff
@@ -7,8 +8,24 @@ class DataModel(BaseModel):
 
 
 class DataCoder:
+    def __init__(self):
+        self.storage = {}
+
```

Comment:
let's extract the dictionary as a Storage, and inject it in `__init__`

```suggestion
    def __init__(self, storage: Storage):
        self.storage = storage

```
In that case we'll need to implement `Storage`
let's do it as simple as possible:
```python
class Storage(dict):
    pass
```

Also, you'll need to fix tests:
- add a fixture for `Storage`
- use this fixture to init `data_coder` fixture